lib/FusionInventory/Agent/Task/Inventory/Win32/Firewall.pm

```diff
@@ -0,0 +1,201 @@
+        "HKEY_LOCAL_MACHINE/SYSTEM/CurrentControlSet/services/SharedAccess/Parameters/FirewallPolicy"
+    );
+    return unless $key;
+    my $subKeys = {
+        domain   => 'DomainProfile',
+        public   => 'PublicProfile',
+        standard => 'StandardProfile'
+    };
+    my $profiles = {};
+    for my $profile (keys %$subKeys) {
+        next unless $key->{$subKeys->{$profile} . '/'};
+        next unless defined $key->{$subKeys->{$profile} . '/'}->{'/EnableFirewall'};
+        my $enabled = hex2dec($key->{ $subKeys->{$profile} . '/'}->{'/EnableFirewall'});
+        $profiles->{$profile} = {
+            STATUS => $enabled
+                ? FIREWALL_STATUS_ON
+                : FIREWALL_STATUS_OFF,
+            PROFILE => $subKeys->{$profile}
+        };
+    }
+
+    return $profiles;
+}
+
+sub _makeProfileAndConnectionsAssociation {
+    my (%params) = @_;
+
+    return unless $params{firewallProfiles};
+
+    my ($profilesKey, $signaturesKey) = _retrieveProfilesAndSignaturesKey();
+    return unless $profilesKey && $signaturesKey;
+
+    foreach my $interface (getInterfaces(
+        additionalPropertiesNetWorkAdapterConfiguration => [qw/DNSDomain/],
+        additionalPropertiesNetWorkAdapter => [qw/GUID/]
+    )) {
+        next if ($interface->{STATUS} ne 'Up');
+
+        my $profile;
+
+        my $domainSettings = _getConnectionDomainSettings(
+            guid => $interface->{GUID}
+        );
+        # check if connection with domain
+        if ($domainSettings) {
+            $profile = _retrieveFirewallProfileWithdomain(
+                profileName => $domainSettings->{'/PrimaryDomainName'},
+                profilesKey => $profilesKey
+            )
+        } else {
+            $profile = _retrieveFirewallProfileWithoutDomain(
+                DNSDomain => $interface->{DNSDomain},
+                profilesKey => $profilesKey,
+                signaturesKey => $signaturesKey
+            );
+        }
+
+        next unless $profile;
+	
+        my $category = hex2dec($profile->{'/Category'});
+        if (not defined $params{firewallProfiles}->{$mappingFirewallProfiles[$category]}->{CONNECTIONS}) {
+            $params{firewallProfiles}->{$mappingFirewallProfiles[$category]}->{CONNECTIONS} = [];
+        }
+        my $connection = {DESCRIPTION => $interface->{DESCRIPTION}};
+        $connection->{IPADDRESS} = $interface->{IPADDRESS} if ($interface->{IPADDRESS});
+        $connection->{IPADDRESS6} = $interface->{IPADDRESS6} if ($interface->{IPADDRESS6});
+        push @{$params{firewallProfiles}->{$mappingFirewallProfiles[$category]}->{CONNECTIONS}}, $connection;
+    }
+
+    my @profiles = ();
+    for my $p (values %{$params{firewallProfiles}}) {
+        my @p;
+        if ($p->{CONNECTIONS} && ref($p->{CONNECTIONS}) eq 'ARRAY') {
+            my @conns = @{$p->{CONNECTIONS}};
+            delete $p->{CONNECTIONS};
+            for my $conn (@conns) {
+                my $newP = dclone $p;
```

Comment:
I don't absolutely need dclone but as it's already used by project, I preferred this way to clone the variable instead of doing a loop...